internal/pkg/source/source.go

```diff
@@ -114,7 +115,7 @@ func sourcesFromClosures(fn *ssa.Function, conf *config.Config, taggedFields fie
 }
 
 // sourcesFromBlocks finds Source values created by instructions within a function's body.
-func sourcesFromBlocks(fn *ssa.Function, conf *config.Config, taggedFields fieldtags.ResultType) []*Source {
+func sourcesFromBlocks(fn *ssa.Function, conf *config.Config, taggedFields fieldtags.ResultType, propagators fieldpropagator.ResultType) []*Source {
```

Comment:
Some of these signatures are starting to get heavy. It's also kind of annoying to have to pipe all of these arguments down several calls. I think it would make sense to bundle `conf`, `taggedFields` and `propagators` into some kind of unifying abstraction. Maybe something like... `Classifier`? :grimacing:

I think a good way to do it would be to embed `config.Config`, `fieldtags.ResultType` and `fieldpropagator.ResultType` into a new struct type. There may be name clashes that we would need to resolve, though.

I think this change could be done in a future PR.

WDYT?

Reply:
Yeah... I think "upstream analyzer stuff" could be a good unit of abstraction - config + any previous analyzer `Result`s.  Of course, at that point we could just pass config and `pass` to every function.

On the one hand, it feels like a code-smell reliance on globals.  I suppose globals aren't inherently evil, particularly when they're static, so this might be an unwarranted knee-jerk aversion to it.

But yes, a future refactor PR might be a better place for it.